For static linking, we need to use -l<libname> and not <full path to lib>,
else it tries to link against the .so.  The lib-link.m4 code is limited here.

Martin Schlemmer <azarah@gentoo.org>

--- configure
+++ configure
@@ -5862,7 +5862,7 @@ fi
                         LTLIBICONV="${LTLIBICONV}${LTLIBICONV:+ }-L$found_dir -l$name"
             if test "X$found_so" != "X"; then
                                                         if test "$enable_rpath" = no || test "X$found_dir" = "X/usr/$acl_libdirstem"; then
-                                LIBICONV="${LIBICONV}${LIBICONV:+ }$found_so"
+                                LIBICONV="${LIBICONV}${LIBICONV:+ }-l$name"
               else
                                                                                 haveit=
                 for x in $ltrpathdirs; do
@@ -5875,10 +5875,10 @@ fi
                   ltrpathdirs="$ltrpathdirs $found_dir"
                 fi
                                 if test "$acl_hardcode_direct" = yes; then
-                                                      LIBICONV="${LIBICONV}${LIBICONV:+ }$found_so"
+                                                      LIBICONV="${LIBICONV}${LIBICONV:+ }-l$name"
                 else
                   if test -n "$acl_hardcode_libdir_flag_spec" && test "$acl_hardcode_minus_L" = no; then
-                                                            LIBICONV="${LIBICONV}${LIBICONV:+ }$found_so"
+                                                            LIBICONV="${LIBICONV}${LIBICONV:+ }-l$name"
                                                             haveit=
                     for x in $rpathdirs; do
                       if test "X$x" = "X$found_dir"; then
@@ -5910,7 +5910,7 @@ fi
                       LIBICONV="${LIBICONV}${LIBICONV:+ }-L$found_dir"
                     fi
                     if test "$acl_hardcode_minus_L" != no; then
-                                                                                        LIBICONV="${LIBICONV}${LIBICONV:+ }$found_so"
+                                                                                        LIBICONV="${LIBICONV}${LIBICONV:+ }-l$name"
                     else
                                                                                                                                                                                 LIBICONV="${LIBICONV}${LIBICONV:+ }-l$name"
                     fi
@@ -6854,7 +6854,7 @@ fi
                         LTLIBINTL="${LTLIBINTL}${LTLIBINTL:+ }-L$found_dir -l$name"
             if test "X$found_so" != "X"; then
                                                         if test "$enable_rpath" = no || test "X$found_dir" = "X/usr/$acl_libdirstem"; then
-                                LIBINTL="${LIBINTL}${LIBINTL:+ }$found_so"
+                                LIBINTL="${LIBINTL}${LIBINTL:+ }-l$name"
               else
                                                                                 haveit=
                 for x in $ltrpathdirs; do
@@ -6867,10 +6867,10 @@ fi
                   ltrpathdirs="$ltrpathdirs $found_dir"
                 fi
                                 if test "$acl_hardcode_direct" = yes; then
-                                                      LIBINTL="${LIBINTL}${LIBINTL:+ }$found_so"
+                                                      LIBINTL="${LIBINTL}${LIBINTL:+ }-l$name"
                 else
                   if test -n "$acl_hardcode_libdir_flag_spec" && test "$acl_hardcode_minus_L" = no; then
-                                                            LIBINTL="${LIBINTL}${LIBINTL:+ }$found_so"
+                                                            LIBINTL="${LIBINTL}${LIBINTL:+ }-l$name"
                                                             haveit=
                     for x in $rpathdirs; do
                       if test "X$x" = "X$found_dir"; then
@@ -6902,7 +6902,7 @@ fi
                       LIBINTL="${LIBINTL}${LIBINTL:+ }-L$found_dir"
                     fi
                     if test "$acl_hardcode_minus_L" != no; then
-                                                                                        LIBINTL="${LIBINTL}${LIBINTL:+ }$found_so"
+                                                                                        LIBINTL="${LIBINTL}${LIBINTL:+ }-l$name"
                     else
                                                                                                                                                                                 LIBINTL="${LIBINTL}${LIBINTL:+ }-l$name"
                     fi
@@ -26907,7 +26907,7 @@ fi
                         LTLIBGNUTLS="${LTLIBGNUTLS}${LTLIBGNUTLS:+ }-L$found_dir -l$name"
             if test "X$found_so" != "X"; then
                                                         if test "$enable_rpath" = no || test "X$found_dir" = "X/usr/$acl_libdirstem"; then
-                                LIBGNUTLS="${LIBGNUTLS}${LIBGNUTLS:+ }$found_so"
+                                LIBGNUTLS="${LIBGNUTLS}${LIBGNUTLS:+ }-l$name"
               else
                                                                                 haveit=
                 for x in $ltrpathdirs; do
@@ -26920,10 +26920,10 @@ fi
                   ltrpathdirs="$ltrpathdirs $found_dir"
                 fi
                                 if test "$acl_hardcode_direct" = yes; then
-                                                      LIBGNUTLS="${LIBGNUTLS}${LIBGNUTLS:+ }$found_so"
+                                                      LIBGNUTLS="${LIBGNUTLS}${LIBGNUTLS:+ }-l$name"
                 else
                   if test -n "$acl_hardcode_libdir_flag_spec" && test "$acl_hardcode_minus_L" = no; then
-                                                            LIBGNUTLS="${LIBGNUTLS}${LIBGNUTLS:+ }$found_so"
+                                                            LIBGNUTLS="${LIBGNUTLS}${LIBGNUTLS:+ }-l$name"
                                                             haveit=
                     for x in $rpathdirs; do
                       if test "X$x" = "X$found_dir"; then
@@ -26955,7 +26955,7 @@ fi
                       LIBGNUTLS="${LIBGNUTLS}${LIBGNUTLS:+ }-L$found_dir"
                     fi
                     if test "$acl_hardcode_minus_L" != no; then
-                                                                                        LIBGNUTLS="${LIBGNUTLS}${LIBGNUTLS:+ }$found_so"
+                                                                                        LIBGNUTLS="${LIBGNUTLS}${LIBGNUTLS:+ }-l$name"
                     else
                                                                                                                                                                                 LIBGNUTLS="${LIBGNUTLS}${LIBGNUTLS:+ }-l$name"
                     fi
@@ -27183,7 +27183,7 @@ if test "${ac_cv_libgnutls+set}" = set; then
 else
 
     ac_save_LIBS="$LIBS"
-    LIBS="$LIBS $LIBGNUTLS"
+    LIBS="$LIBGNUTLS $LIBS"
     cat >conftest.$ac_ext <<_ACEOF
 /* confdefs.h.  */
 _ACEOF
@@ -27636,7 +27636,7 @@ fi
                         LTLIBSSL="${LTLIBSSL}${LTLIBSSL:+ }-L$found_dir -l$name"
             if test "X$found_so" != "X"; then
                                                         if test "$enable_rpath" = no || test "X$found_dir" = "X/usr/$acl_libdirstem"; then
-                                LIBSSL="${LIBSSL}${LIBSSL:+ }$found_so"
+                                LIBSSL="${LIBSSL}${LIBSSL:+ }-l$name"
               else
                                                                                 haveit=
                 for x in $ltrpathdirs; do
@@ -27649,10 +27649,10 @@ fi
                   ltrpathdirs="$ltrpathdirs $found_dir"
                 fi
                                 if test "$acl_hardcode_direct" = yes; then
-                                                      LIBSSL="${LIBSSL}${LIBSSL:+ }$found_so"
+                                                      LIBSSL="${LIBSSL}${LIBSSL:+ }-l$name"
                 else
                   if test -n "$acl_hardcode_libdir_flag_spec" && test "$acl_hardcode_minus_L" = no; then
-                                                            LIBSSL="${LIBSSL}${LIBSSL:+ }$found_so"
+                                                            LIBSSL="${LIBSSL}${LIBSSL:+ }-l$name"
                                                             haveit=
                     for x in $rpathdirs; do
                       if test "X$x" = "X$found_dir"; then
@@ -27684,7 +27684,7 @@ fi
                       LIBSSL="${LIBSSL}${LIBSSL:+ }-L$found_dir"
                     fi
                     if test "$acl_hardcode_minus_L" != no; then
-                                                                                        LIBSSL="${LIBSSL}${LIBSSL:+ }$found_so"
+                                                                                        LIBSSL="${LIBSSL}${LIBSSL:+ }-l$name"
                     else
                                                                                                                                                                                 LIBSSL="${LIBSSL}${LIBSSL:+ }-l$name"
                     fi
@@ -27912,7 +27912,7 @@ if test "${ac_cv_libssl+set}" = set; then
 else
 
     ac_save_LIBS="$LIBS"
-    LIBS="$LIBS $LIBSSL"
+    LIBS="$LIBSSL $LIBS"
     cat >conftest.$ac_ext <<_ACEOF
 /* confdefs.h.  */
 _ACEOF
